### PR TITLE
build: add check-manifest, update `MANIFEST.in`

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

This adds [check-manifest](https://github.com/mgedmin/check-manifest) to ensure that the files checked in to git match the list of files in the built sdist, and that no files/packages are missing or included unnecessarily.
`MANIFEST.in` was updated accordingly, also fixing the `sdist -> wheel` build step that was broken due to a change in #529, omitting the `requirements/requirements_*.txt` files (see also https://github.com/pypa/build/issues/332).

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -143,3 +143,23 @@ exclude_lines = [
...
+    "CONTRIBUTING.md",
+    "assets/**",
+    "changelog/**",
+    "docs/**",
+    "examples/**",
+    # tests
+    "test_bot/**",
+    "test_launcher.py",
+    "tests/**",
+]
```
Comment (onerandomusername) [inline pyproject.toml:165]: Part of the reason for a manifest is to include the source. I feel like we should include a few of these files we're leaving off.

https://discuss.python.org/t/purpose-of-an-sdist/4639

**Review by onerandomusername — CHANGES_REQUESTED** [review]
Some thoughts.

File: pyproject.toml (hunk 1 above)
Comment (shiftinv) [reply pyproject.toml:165]: There seem to be very differing opinions about this, quoting a few people in that thread:

> So it does not (need to) include any files that aren’t relevant for installing the package, but it also should not contain platform-specific binaries.

> An sdist should also include tests and documentation

> And this is what sdists are to me: whatever is required to build a wheel.

I personally agree with the notion of "an sdist should include everything required to install the package/build a wheel"; the sdist size would increase quite a bit if we were to include documentation (+ images) and tests. The full source including everything is already available from github, which is also linked in the package metadata.

There's also [this](https://discuss.python.org/t/should-sdists-include-docs-and-tests/14578/) post with similar opinions, but ultimately I'd opt for keeping the status quo for now.

**Review by onerandomusername — CHANGES_REQUESTED** [review]
I used `python -m build` to test the behavior of this and I noticed a deprecation warning from setuptools about `disnake.bin` -- we should add `"disnake.bin"` to the list of packages in setup.py.

**Comment by shiftinv** [comment]
> I used `python -m build` to test the behavior of this and I noticed a deprecation warning from setuptools about `disnake.bin` -- we should add `"disnake.bin"` to the list of packages in setup.py.

Ah, good catch, fixed in https://github.com/DisnakeDev/disnake/pull/674/commits/2929c4c00964b731a4446f4c27edb76778787a9f.

File: .github/workflows/lint-test.yml [hunk 2]
```diff
@@ -131,13 +131,19 @@ jobs:
...
         nox-setup: tools
 
     - name: Run slotscheck
-      run: task slotscheck
+      if: (success() || failure()) && steps.setup.outcome == 'success'
+      run: nox -s slotscheck
+
+    - name: Run check-manifest
+      if: (success() || failure()) && steps.setup.outcome == 'success'
+      run: nox -s check-manifest
```
Comment (onerandomusername) [inline .github/workflows/lint-test.yml:147]: perhaps after this step we should run `python -m build` to test that the project properly builds..?

Comment (shiftinv) [reply .github/workflows/lint-test.yml:147]: Once we have a proper build+release workflow we should use that instead, but for now I've added a build step in https://github.com/DisnakeDev/disnake/pull/674/commits/a57887200392f62b993a705d1622af8057098a84.

**Review by onerandomusername — APPROVED** [review]
looks good, we can argue over what to put in the sdist some other time 😄

**Pull request merged**